### PR TITLE
Code

### PR DESCRIPTION
**Pull request by andysim3d**
implement code block and parser.

### CODE REVIEW

File: codeblock_test.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    ('print(hello `Andy`)', '<code>print(hello `Andy`)</code>'),
+])
+def test_render(content, html):
+    fcb = CodeBlock(content)
+    assert html == fcb.render()
+
+
+@pytest.mark.parametrize("content, expected_html, start, end", [
+    ('`code`', '<code>code</code>', 0, 6),
+<<<<<<< HEAD
```
Comment (jinglongwu) [inline codeblock_test.py]: somehow conflicts

File: codeblock.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    def _render(self):
+        return '<code>{}</code>'.format(self.content())
+
+    @staticmethod
+    def parse(content: Text):
+        '''
+        Parse a text content, and return (begin, end, Element) if parseable,
+        or (-1, -1, None) that no such a element
+        '''
+        code_pattern = r"(^`|[^\\]`)((([^\n])*)[^\\])*`"
```
Comment (jinglongwu) [inline codeblock.py]: if you don't need group(3) and group(4), the regex can be simplified as 
```python
r"(^`|[^\\]`)([^\n]*[^\\])*`"
```

also, maybe the last asterisk is not necessary? What you want is make sure there is no back-slash before back-tick, then maybe this is good enough?
```python
r"(^`|[^\\]`)([^\n]*[^\\])`"
```

Comment (andysim3d) [reply codeblock.py]: That's necessary. without that, it can not parse ``, cause nothing inside code block.

**Review by jinglongwu — APPROVED** [review]
Looks good to me

**Pull request merged**